docs/data-structures/binary-search-tree.mdx

```diff
@@ -0,0 +1,483 @@
+---
+title: Binary Search Tree
+slug: /data-structures/binary-search-tree
+---
+
+## Introduction
+
+The binary search tree is a **binary tree** in which for every node, all the elements in its left sub-tree are smaller than that node, and all the elements in its right sub-tree are greater than that node. A binary search tree is useful for searching an element in less number of comparisons. Binary search trees are considered to be efficient data structures because they speed up the process of **insertion** and **deletion**.
+
+## Explanation
+
+<p align="center">
+  <img width="500" height="300" src="https://i.imgur.com/aYWkuCT.png" />
+</p>
+
+As you can see in the above diagram, The `root` node is `30`. The `left sub-tree` of the root node consists of nodes `10`, `15` and `20`. All these nodes have `lesser values` than the root node. The `right sub-tree` of the root node consists of nodes `40`, `50` and `60`. All these nodes have `greater values` than the root node. Each of the sub-trees recursively obeys the binary search tree constraint. For example, in the left sub-tree of the root node, `15` is the root and the element in its left sub-tree `(10)` is smaller than 15, while the node present in the right sub-tree `(20)` is greater than the root node's value.
```

Comment:
No need to keep `left sub-tree in backticks. No need to keep `right sub-tree` in backticks. Keep `lesser values` and `greater values` in bold